definition_validator/validator.cc

```diff
@@ -530,9 +530,57 @@ void validateUnsatisfiedRequiredAncestors(core::Context ctx, const core::SymbolR
     }
 }
 
+void validateUnsatisfiableClassAncestors(core::Context ctx, const core::SymbolRef sym) {
+    auto data = sym.data(ctx);
+
+    vector<core::Symbol::RequiredAncestor> requiredClasses;
+    for (auto ancst : data->requiredAncestorsTransitive(ctx)) {
+        if (ancst.symbol.data(ctx)->isClassOrModuleClass()) {
+            requiredClasses.emplace_back(ancst);
+        }
+    }
+
+    if (data->isClassOrModuleAbstract()) {
+        for (auto ancst : requiredClasses) {
+            if (!sym.data(ctx)->derivesFrom(ctx, ancst.symbol) && !ancst.symbol.data(ctx)->derivesFrom(ctx, sym)) {
+                if (auto e = ctx.state.beginError(data->loc(), core::errors::Resolver::UnsatisfiableRequiredAncestor)) {
+                    e.setHeader("`{}` requires unrelated class `{}` making it impossible to inherit", data->show(ctx),
+                                ancst.symbol.data(ctx)->show(ctx));
+                    e.addErrorLine(ancst.loc, "`{}` is required by `{}` here", ancst.symbol.data(ctx)->show(ctx),
+                                   ancst.origin.data(ctx)->show(ctx));
+                }
+            }
+        }
+    }
+
+    if (requiredClasses.size() <= 1) {
+        return;
+    }
+
+    for (int i = 0; i < requiredClasses.size() - 1; i++) {
+        auto ra1 = requiredClasses[i];
+        for (int j = i + 1; j < requiredClasses.size(); j++) {
+            auto ra2 = requiredClasses[j];
+            if (ra1.symbol != ra2.symbol && !ra1.symbol.data(ctx)->derivesFrom(ctx, ra2.symbol) &&
```

Comment:
Isn't it alway the case that `ra1.symbol != ra2.symbol`, as `requiredAncestorsTransitive` returns a vector of unique symbols?